apps/charterafrica/src/lib/openAfrica.js

```diff
@@ -2,16 +2,28 @@ import fetchJson from "@/charterafrica/utils/fetchJson";
 
 const BASE_DOCUMENTS_URL = "https://openafrica.net/api/3/action/";
 
-async function formatDatasets(data) {
-  const {
-    result: {
-      count,
-      facets: { tags: allTags, groups: allCountries },
-      results,
-    },
-  } = data || {};
-
-  const formattedDatasets = results?.map((dataset) => {
+const PageSize = 10;
+
+async function packageSearch(params) {
+  try {
+    const response = await fetchJson.get(
+      `${BASE_DOCUMENTS_URL}package_search`,
+      {
+        params,
+      }
+    );
+    return response;
+  } catch (error) {
+    return error;
+  }
+}
+
+function formatDatasets(data) {
+  const { result: { count, facets: { tags, groups }, results } = {} } =
+    data || {};
+
+  const documents = [];
+  const datasets = results?.map((dataset) => {
```

Comment:
There has to be a cleaner way of handling this part... may be a standalone `formatDataset` function or something.